pkg/proxy/policy/policy.go

```diff
@@ -24,51 +24,57 @@ func RegisterRule(rulename string, constructor func() Rule) {
 // Policy contains a list of Rules that will be applied to the request.
 type Policy struct {
 	// AnyOf expects incoming requests to satisfy one of these Rules.
-	AnyOf []Rule
+	AnyOf []Rule `json:"anyOf"`
 }
 
 // UnmarshalJSON implements the json.Unmarshaler interface for the Policy class.
 // Expects rule_type to specify an existing rule in the rule registry.
 func (p *Policy) UnmarshalJSON(data []byte) error {
 	var policywrapper struct {
 		Policy struct {
-			AnyOf []json.RawMessage `json:"AnyOf"`
-		} `json:"Policy"`
+			AnyOf []json.RawMessage
+		}
 	}
 	if err := json.Unmarshal(data, &policywrapper); err != nil {
 		return err
 	}
+	rules, err := unmarshalRuleType(policywrapper.Policy.AnyOf)
+	if err != nil {
+		return err
+	}
+	p.AnyOf = rules
+	return nil
+}
 
-	for _, rule := range policywrapper.Policy.AnyOf {
-		var tmpmap map[string]interface{}
+// unmarshalRule tries to unmarshal a rule json into the registered rule types.
```

Comment:
fix docs

Reply:
I just renamed the function to make it more clear what it is trying to do.